packages/runtime/src/components/page/PageHead.tsx

```diff
@@ -0,0 +1,294 @@
+        // Ignore errors from user input.
+      }
+    })
+
+    previousHeadSnippets.current = headSnippets
+  }, [headSnippets])
+
+  const scriptRefs = useRef<(HTMLScriptElement | null)[]>([])
+  function saveScriptRef(ref: HTMLScriptElement | null) {
+    if (ref == null) return
+    scriptRefs.current.push(ref)
+  }
+
+  const headElementsFragment = (
+    <>
+      <style data-makeswift-meta="true">
+        {`
+        html {
+            font-family: sans-serif;
+        }
+        div#__next {
+            overflow: hidden;
+        }
+        `}
+      </style>
+      {favicon && (
+        <link
+          rel="icon"
+          type={favicon.mimetype}
+          href={favicon.publicUrl}
+          data-makeswift-meta="true"
+        />
+      )}
+
+      {canonicalUrl && <link rel="canonical" href={canonicalUrl} data-makeswift-meta="true" />}
+
+      {isIndexingBlocked && <meta name="robots" content="noindex" data-makeswift-meta="true" />}
+
+      {title && (
+        <>
+          <title data-makeswift-meta="true">{title}</title>
+          <meta property="og:title" content={title} data-makeswift-meta="true" />
+          <meta name="twitter:title" content={title} data-makeswift-meta="true" />
+          <meta itemProp="name" content={title} data-makeswift-meta="true" />
+        </>
+      )}
+
+      {description && (
+        <>
+          <meta name="description" content={description} data-makeswift-meta="true" />
+          <meta property="og:description" content={description} data-makeswift-meta="true" />
+          <meta name="twitter:description" content={description} data-makeswift-meta="true" />
+          <meta itemProp="description" content={description} data-makeswift-meta="true" />
+        </>
+      )}
+
+      {keywords && <meta name="keywords" content={keywords} data-makeswift-meta="true" />}
+
+      {socialImage && (
+        <>
+          <meta property="og:image" content={socialImage.publicUrl} data-makeswift-meta="true" />
+          <meta
+            property="og:image:type"
+            content={socialImage.mimetype}
+            data-makeswift-meta="true"
+          />
+          <meta name="twitter:image" content={socialImage.publicUrl} data-makeswift-meta="true" />
+          <meta name="twitter:card" content="summary_large_image" data-makeswift-meta="true" />
+          <meta itemProp="image" content={socialImage.publicUrl} data-makeswift-meta="true" />
+        </>
+      )}
+
+      {fontFamilyParamValue !== '' && (
+        <link
+          rel="stylesheet"
+          href={`https://fonts.googleapis.com/css?family=${fontFamilyParamValue}&display=swap`}
+          data-makeswift-meta="true"
+        />
+      )}
+
+      {headSnippets.map(snippetToElement).map(children =>
+        Children.map(children, child => {
+          if (typeof child === 'string') return child
+
+          if ((child.type as string) === 'script') {
+            return cloneElement(child, { ref: saveScriptRef })
```

Comment:
After reading the rest of this file I would recommend a different approach here. Instead of using `cloneElement` and adding this ref, I would leverage `findDOMNode` to get all DOM nodes that are put in a portal for CSN. From there, you can traverse the DOM and execute any scripts as necessary.

Reply:
Based on my understanding of the docs, `findDOMNode` is deprecated and only for use with class components: https://react.dev/reference/react-dom/findDOMNode#finddomnode. Wouldn't it be worse to rely on this?

Reply:
We rely on `findDOMNode` in the runtime to get DOM nodes from user-registered components since they don't always use `forwardRef`. It's been deprecated for a while but hasn't been removed because it's the only way to do what it does.

I guess `cloneElement` plus a ref isn't that bad when compared to `findDOMNode`.